htlcswitch/link.go

```diff
@@ -2419,6 +2472,49 @@ func (l *channelLink) handleUpstreamMsg(msg lnwire.Message) {
 
 }
 
+// drivePendingStfuSends checks to see if we owe any Stfu messages to our peer.
+// If we do then we will send them and record that send. We can owe our peer an
+// Stfu either if we have a pending request to quiesce the channel or our peer
+// has sent us an Stfu to which we haven't yet responded.
+func (l *channelLink) drivePendingStfuSends() {
+	if !l.quiescer.oweStfu() {
```

Comment:
Perhaps this can use the existing hooks in place to automatically call this method once the `quiescer` decides that we owe a message.

Reply:
The existing hooks only kick in once all the htlcs have cleared, we would need to introduce new hooks, which I don't mind but the condition we are looking for here is whether all local updates have been dually signed.